checklist/checklist_sys.csv/tests.csv

```diff
@@ -0,0 +1,9 @@
+ID,Topic,Title,Requirement,Explanation,References
+2.1,Data Presence,Test Data Fetching and File Reading,"Verify that the data fetching API or data file reading functionality works correctly. Ensure that proper error handling is in place for scenarios such as missing files, incorrect file formats, and network errors.","Ensure that the code responsible for fetching or reading data can handle errors. This means if the file is missing, the format is wrong, or there's a network issue, the system should not crash but should provide a clear error message indicating the problem.",(general knowledge)
+3.1,Data Quality,Validate Data Shape and Values,"Check that the data has the expected shape and that all values meet domain-specific constraints, such as non-negative distances.","Check that the data being used has the correct structure (like having the right number of columns) and that the values within the data make sense (e.g., distances should not be negative). This ensures that the data is valid and reliable for model training.","alexander2024Evaluating, ISO/IEC5259"
+3.2,Data Quality,Check for Duplicate Records in Data,Check for duplicate records in the dataset and ensure that there are none.,"Ensure that the dataset does not contain duplicate entries, as these can skew the results and reduce the model's performance. The test should identify any repeated records so they can be removed or investigated.",ISO/IEC5259
+4.1,Data Ingestion,Verify Data Split Proportion,Check that the data is split into training and testing sets in the expected proportion.,"Confirm that the data is divided correctly into training and testing sets according to the intended ratio. This is crucial for ensuring that the model is trained and evaluated properly, with representative samples in each set.","openja2023studying, DBLP:conf/recsys/Kula15, singh2020mmf"

```

Comment:
So here, in a data splitting script, I think I would expect at least three functions:

1. A function to read in the raw data that exists on disc.
2. A function to split the data into at least a training and test set (maybe a training, validation and test set). This function might also stratify across important variables (most often the target, but there can be others) as well as group data for appropriate splitting of non-independent observations (e.g., in the case of time series or spatial data).
3. A function that writes the raw trained data and raw test data to disc.

Oftentimes such a script at this point in the ML pipeline would also create a pre-processor (scikit-learn) pipeline object from the training data, which would also be saved to disc. It is useful to do this early on, so that one can pre-process data to be used for EDA, as well as having it available to read into later training and test/prediction scripts.

Reply:
As the functions for building the ML model, I would expect the 3 function you mentioned
As a function for unit case, I can imagine a test function that read in train & test data (e.g. from `data/processed/`) and verify if the size ratio is as expected (e.g. `train_data.shape[0]/test_data.shape[0] == 0.8/0.2`)